File: Good, good Python/step 06/6.03/6.3.05.py
'''

Подвиг 7. Вводятся целые числа в одну строку через пробел. На их основе формируется кортеж. 
Необходимо создать еще один кортеж с уникальными (не повторяющимися) значениями из первого кортежа. Результат отобразите в виде списка чисел через пробел.

P. S. Подобные задачи решаются, как правило, с помощью множеств, но в качестве практики пока обойдемся без них.

Sample Input:

8 11 -5 -2 8 11 -5

Sample Output:

8 11 -5 -2

'''

old_tuple = tuple(map(int, input().split()))
new_tuple = ()
for i in old_tuple:
    if i not in new_tuple:
        new_tuple += i,
print(*tuple(new_tuple))

File: Fundamentals of C programming/step06/6.3/6.3.05.c
/*
Кнопочный электронный кодовый замок имеет десять кнопок. Каждая из кнопок имеет свой порядковый номер от 0 до 9. Правильный код 1024 зашит внутрь замка. Человек, который хочет открыть дверь, должен ввести на циферблате последовательно 1, 0, 2 и 4. Напишите программу, моделирующую работу такого замка.

Входные данные:
Четыре целых числа b1,b2,b3,b4b1​,b2​,b3​,b4​ -- номера кнопок, которые нажал человек.

Выходные данные:
Строка open, если введён правильный код. Строка close, если введён неправильный код.


Sample Input 1:

1 0 2 4

Sample Output 1:

open

Sample Input 2:

1 0 2 3

Sample Output 2:

close

*/

#include <stdio.h>

int main() {
  int b1,b2,b3,b4;
  scanf("%d %d %d %d",&b1,&b2,&b3,&b4);
    
  if (b1 == 1 && b2 == 0 && b3 == 2 && b4 == 4) {
	  printf("open");
     } else 
	   printf("close");
    
  return 0;
}
